substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -704,16 +667,21 @@ pub mod pallet {
 	/// whole era. For this reason they are kept here - only staking pallet knows about eras. The
 	/// implementor of [`DisablingStrategy`] defines if a validator should be disabled which
 	/// implicitly means that the implementor also controls the max number of disabled validators.
-	///
+
 	/// The vec is always kept sorted based on the u32 index so that we can find whether a given
 	/// validator has previously offended using binary search.
 	///
 	/// Additionally, each disabled validator is associated with an `OffenceSeverity` which
 	/// represents how severe is the offence that got the validator disabled.
 	#[pallet::storage]
-	#[pallet::unbounded]
-	pub type DisabledValidators<T: Config> =
-		StorageValue<_, Vec<(u32, OffenceSeverity)>, ValueQuery>;
+	pub type DisabledValidators<T: Config> = StorageValue<
+		_,
+		BoundedVec<
+			(u32, OffenceSeverity),
+			<T::ElectionProvider as ElectionProviderBase>::MaxWinners,
```

Comment:
Same as above.

Also, consider using `WeaklyBoundedVec` since we expect validator numbers to change over time. An increase in the value is not an issue but for some reason if we decrease it, it can lead to decode issues with `BoundedVec`.

Reply:
Actually, `DisabledValidators` would be a much smaller number than the actual validator set, and it depends on the `DisablingStrategy`, which is currenctly 1/3rd of the set. This can also be a new config value.

Reply:
Using a different `MaxDisabledValidators` constant